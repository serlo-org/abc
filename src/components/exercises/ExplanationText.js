import React from 'react';
import { View, Text } from 'react-native';

import speakerImage from '../../assets/images/speaker.png';
import { loadSound } from '../helpers/audio';
import { RoundButton } from '../Components';

const styles = {
  container: {
    flex: 1,
    backgroundColor: 'rgb(181,206,77)',
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  text: {
    color: '#FFF',
    fontSize: 40,
    fontWeight: 'bold',
    textAlign: 'center'
  }
};

const ExplanationText = ({ text, sound }) => {
  const play = () => {
    sound.playAsync();
    sound.setPlaybackFinishedCallback(() => {
      sound.setPositionAsync(0);
    });
  };

  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        {text}
      </Text>
      <RoundButton icon={speakerImage} size={40} onPress={play} />
    </View>
  );
};

export default loadSound(ExplanationText);
