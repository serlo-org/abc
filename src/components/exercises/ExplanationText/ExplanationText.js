import { Ionicons } from '@expo/vector-icons';
import React from 'react';
import { View, Text } from 'react-native';

import { DEFAULT } from '../../../styles/text';
import { PlaySounds } from '../../helpers/PlaySounds';
import { RoundIconButton } from '../../common/buttons';

const styles = {
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around'
  },
  text: DEFAULT
};

const ExplanationText = ({ text, sound, setState }) => (
  <PlaySounds
    playInitially
    onPlayEnd={() => setState(true)}
    sounds={[sound]}
    render={buttonProps => (
      <View style={styles.container}>
        <Text style={styles.text}>{text}</Text>
        <RoundIconButton
          IconComponent={Ionicons}
          name="md-volume-up"
          size={40}
          {...buttonProps}
        />
      </View>
    )}
  />
);

export default ExplanationText;
