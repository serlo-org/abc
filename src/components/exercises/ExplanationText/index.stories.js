import { storiesOf } from '@storybook/react-native';

import * as ExplanationText from '.';
import { createStoriesFromFixtures } from '../helpers';

const story = storiesOf('exercises/ExplanationText', module);

createStoriesFromFixtures(story, ExplanationText);
