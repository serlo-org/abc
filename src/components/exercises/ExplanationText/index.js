import { InfoScreen } from '../../../exercises';
import Component from './ExplanationText';

export const Exercise = InfoScreen.Exercise;
export const fixtures = InfoScreen.createFixtures([
  {
    name: 'Wiederholen Sie den Buchstaben.',
    props: {
      text: 'Wiederholen Sie den Buchstaben.',
      sound: require('../../../assets/sounds/exercises/wiederholen_sie_den_buchstaben_a.mp3')
    }
  }
]);
export { Component };
