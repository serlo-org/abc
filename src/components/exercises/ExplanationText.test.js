import React from 'react';

import ExplanationText from './ExplanationText';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const tree = renderer.create(
    <ExplanationText
      text="Wiederholen Sie den Buchstaben."
      sound={require('../../assets/sounds/exercises/wiederholen_sie_den_buchstaben_a.mp3')}
    />
  );
  expect(tree).toMatchSnapshot();
});
