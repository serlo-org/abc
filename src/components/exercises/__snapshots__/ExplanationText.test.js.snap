// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(181,206,77)",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "fontSize": 40,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Wiederholen Sie den Buchstaben.
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "#73DBFF",
          "borderRadius": 9999,
          "elevation": 10,
          "padding": 5,
          "shadowColor": "rgba(0, 0, 0, 0.1)",
          "shadowOffset": Object {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 40,
            "resizeMode": "contain",
            "width": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;
